src/main/java/duke/Ui.java

```diff
@@ -0,0 +1,69 @@
+package duke;
+
+import java.util.Scanner;
+
+public class Ui {
+    private Scanner scanner;
+
+    /**
+     * Constructs an object that handles display of prompts and receiving user input.
+     */
+    public Ui() {
+        scanner = new Scanner(System.in);
+    }
+
+    /**
+     * Prints a message to the console.
+     *
+     * @param message the message to be printed.
+     */
+    public void printMessage(String message) {
+        String[] messages = message.split("\n");
+        System.out.println("    ____________________________________________________________");
+        for (String line : messages) {
+            System.out.println("     " + line);
+        }
+        System.out.println("    ____________________________________________________________");
+        System.out.println();
+    }
+
+    /**
+     * Prints an error to the console.
+     *
+     * @param e the error to be printed.
+     */
+    public void printError(Exception e) {
+        printMessage("☹ OOPS!!! " + e.getMessage());
+    }
+
+    /**
+     * Reads a command from user input.
+     *
+     * @return the raw command string read from user input.
+     */
+    public String readCommand() {
+        return scanner.nextLine();
+    }
+
+    /**
+     * Shows the welcome message.
+     */
+    public void showWelcome() {
+        printMessage("Hello! I'm Duke\nWhat can I do for you?");
+    }
+
+    /**
+     * Converts the item specified in its plural form.
```

Comment:
The function pluralize() is efficient and does what it needs to. Its placement is slightly odd though as it does not seem to belong in the Ui class. Maybe you could consider shifting it elsewhere more fitting, to the Task/TaskList class for example. Example usage: TaskList.toString() could give "Now you have 2 tasks in the list."